templates/types/streaming/nextjs/app/components/ui/chat/chat-message/markdown.tsx

```diff
@@ -80,6 +117,27 @@ export default function Markdown({ content }: { content: string }) {
             />
           );
         },
+        a({ href, children }) {
+          // If a text link starts with 'citation:', then render it as a citation reference
+          if (
+            Array.isArray(children) &&
+            typeof children[0] === "string" &&
+            children[0].startsWith("citation:")
+          ) {
+            return (
+              <sup>
+                <a
+                  href={href}
```

Comment:
i guess href is not used?
also, better directly render `<SourceNumberButton/>` to reuse the style

Reply:
Oh, i don't know that it can render the React component directly there - updated that. btw, `href` is used for the link of the reference number which should point to the file URL.